cmd/juju/action/call.go

```diff
@@ -308,11 +329,18 @@ func (c *callCommand) Run(ctx *cmd.Context) error {
 		if err != nil {
 			return errors.Trace(err)
 		}
+		if i == 0 {
+			close(done)
```

Comment:
(Also, this method's really long - can you break it up a bit?)

Reply:
The GetActionResult() API call above this blocks until the action is marked as completed. You can only log progress messages to running actions. Closing the done channel here is a means to inform the go routine processing the log messages that they have all been collected and so that go routine can exit